Task5Test/src/test/java/com/company/TriangleTest.java

```diff
@@ -3,75 +3,94 @@
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.File;
 import java.math.BigDecimal;
 
 public class TriangleTest {
 
-    Triangle triangle;
-
-    @DataProvider(name = "ValidIsoscelesTriangle")
-    public Object[][] setValidSidesOfTheIsoscelesTriangle() {
-        return new Object[][] {
-                {BigDecimal.valueOf(10), BigDecimal.valueOf(10), BigDecimal.valueOf(11)},
-                {BigDecimal.valueOf(10), BigDecimal.valueOf(11), BigDecimal.valueOf(10)},
-                {BigDecimal.valueOf(11), BigDecimal.valueOf(10), BigDecimal.valueOf(10)},
-        };
-    }
-
-    @Test(dataProvider = "ValidIsoscelesTriangle")
-    public void testPositiveSetValidSidesOfTheIsoscelesTriangle(BigDecimal a, BigDecimal b, BigDecimal c) throws Exception {
-        triangle = new Triangle(a, b, c);
-        Assert.assertEquals(triangle.definesType(), Triangle.ISOSCELES);
-    }
-
-
-    @DataProvider(name = "setInvalidSides")
-    public Object[][] setNotValidSidesOfTheTriangle() {
-        return new Object[][] {
-                {BigDecimal.valueOf(0), BigDecimal.valueOf(1), BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(0), BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), BigDecimal.valueOf(0)},
+    private static final String VALID_TRIANGLES = "validTriangles";
+    private static final String INVALID_TRIANGLES = "invalidTriangles";
+    private static final String ISOSCELES_NODE = "isosceles";
+    private static final String EQUILATERAL_NODE = "equilateral";
+    private static final String SIDE_A = "a";
+    private static final String SIDE_B = "b";
+    private static final String SIDE_C = "c";
 
-                {BigDecimal.valueOf(-1), BigDecimal.valueOf(1), BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(-1), BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), BigDecimal.valueOf(-1)},
 
-                {null, BigDecimal.valueOf(1), BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), null, BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), null},
-        };
-    }
-
-    @Test(dataProvider = "setInvalidSides", expectedExceptions = Exception.class)
-    public void testNegativeSetNotValidSidesOfTheTriangle(BigDecimal a, BigDecimal b, BigDecimal c) throws Exception {
-        triangle = new Triangle(a, b, c);
-    }
+    Triangle triangle;
 
-    @DataProvider(name = "ValidEquilateralTriangle")
-    public Object[][] setValidSidesOfTheEquilateralTriangle() {
-        return new Object[][]{
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), BigDecimal.valueOf(1)}
-        };
+    @DataProvider(name = "valid triangles from xml")
+    public Object[][] readXMLforPositiveIsosceles() throws Exception {
+        File inputFile = new File("/triangleData.xml");

```

Comment:
Invalid filepath - on linux it will search in root "/" directory that is not correct